site/docs/snapshot-tag-branch.md

```diff
@@ -0,0 +1,148 @@
+Users can create an Iceberg branch for such snapshots to maintain its independent lifecycle.
+
+### Experimental Branch
+
+An experimental branch is useful for many user groups, including:
+
+1. Data scientists and ML researchers can easily create an Iceberg branch to experiment with table data without worrying about polluting the main table snapshot.
+2. Data engineers can perform production AB testing against the experimental branch to ensure the correctness of certain table updates.
+3. Data producers can perform test load in a table in an experimental branch, and then append all the loaded files back to the main branch (similar to Git cherry-pick).
+
+!!!Note
+    Iceberg does not plan to offer a Git-like merge operation through branching.
+    Merging arbitrary changes requires a lot of work to keep track of the intent of the commit and the context. 
+    Merging in a table is actually committing a transaction. The expectation is different from a merge in Git, where the lack of a conflict is the definition of "correct". 
+    In a table, the lack of a file conflict does not mean that the transaction can be committed.
+    In addition, longer transaction lengths from branch-like behavior dramatically increases the likelihood that the transaction could fail.
+    The merge feature would likely be supported through multi-table transaction in the future.
+
+## Snapshot Reference
+
+In version control systems like git, branch and tag are both references of commits.
+In Iceberg, we use a similar concept of **Snapshot Reference** to implement branching and tagging.
+
+Each Iceberg table metadata contains a list of `refs` (references), and a `current-branch` indicating the current branch to use.
+When user creates an Iceberg table, the first commit belongs to the default `main` branch.
+Each snapshot reference has a uniquely identifiable name across all references of a table.
+A snapshot can have multiple references. The exact snapshot reference spec is documented at the [Spec](../spec/#snapshot-reference) page.
+Here we will provide some more explanations to the concepts in snapshot reference.
+
+### Reference Type
+
+There are clearly 2 types of snapshot reference, which are `branch` and `tag`. Their key differences are:
+
+- **New commit**: when a new snapshot is added as a child of a referenced snapshot, tag remains on the old snapshot, but branch reference moves to the child.
+
+- **Retention policy**: retention policy affects all the snapshots in a branch, but only a single tagged snapshot. (More details in the next section)
+
+### Retention Policy
+
+Iceberg offers a [snapshot expiration procedure](../spark-procedures/#expire_snapshots) to clean up snapshots that are not needed to free up storage space.
+Retention policy can be configured both globally and on snapshot reference to provide highly flexible customization to the expiration behavior.
+
+#### Global snapshot retention policy
```

Comment:
I am bit lost here, I thought we will there will be as well a property to disable the retention either on snapshot or globally?

Reply:
we have `GC_ENABLED` table property for that.
more details here: https://github.com/apache/iceberg/issues/3246

Reply:
I think that "global" might be confusing. How about "Table level retention policy" and "Branch level retention policy"? Reference retention (`max-ref-age-ms`) can also be a separate section.